src/arrays.js

```diff
@@ -1,61 +1,75 @@
 const getNthElement = (index, array) => {
-  // your code here
+       if(index >= array.length) {
+      index = index - array.length;
+        return array[index] 
+    }
+
+    else {
+        return array[index]
+	}
 };
 
 const arrayToCSVString = array => {
-  // your code here
+  	return array.toString()
 };
 
 const csvStringToArray = string => {
-  // your code here
+  	return string.split(",")
 };
 
 const addToArray = (element, array) => {
-  // your code here
+  array.push(element)
 };
 
 const addToArray2 = (element, array) => {
-  // your code here
+  newArray = array.concat(element)
+ return newArray
 };
 
 const removeNthElement = (index, array) => {
-  // your code here
+  array.splice(index, 1)
 };
 
 const numbersToStrings = numbers => {
-  // your code here
+  return numbers.map(String)
 };
 
 const uppercaseWordsInArray = strings => {
-  // your code here
+  return strings.map(string => string.toUpperCase());
 };
 
 const reverseWordsInArray = strings => {
-  // your code here
+ // this one had me stumped! I tried many different ways but couldn't figure it out, please help! From what I can see, I believe it is some combination of Split(), reverse() and then join(), maybe it needs to iterate over each?
 };
 
 const onlyEven = numbers => {
-  // your code here
+  return numbers.filter( x => x % 2 === 0);
 };
 
 const removeNthElement2 = (index, array) => {
-  // your code here
+ remArray = array.slice(0, index).concat(array.slice((index + 1)));
+  return remArray 
+
+//this one almost drove me mad :). I didn't think to slice it either side of the index then combine for so long. It seemed counterintuitive for there not to be a way for slice to start after the index and then loop round to before the index!
 };
 
 const elementsStartingWithAVowel = strings => {
-  // your code here
+    const words = strings.filter(string => string.toLowerCase().startsWith("a") || string.toLowerCase().startsWith("e") || string.toLowerCase().startsWith("i") || string.toLowerCase().startsWith("o") || string.toLowerCase().startsWith("u"));
+	return words
 };
 
 const removeSpaces = string => {
-  // your code here
+  return string.split(" ").join("");
 };
 
 const sumNumbers = numbers => {
-  // your code here
+   const reducer = (accumulator, currentValue) => accumulator + currentValue;
+return numbers.reduce(reducer)
```

Comment:
So, we have an array **numbers** with [1, 2, 3, 4, 5];

numbers.reduce((accumulator, currentValue) =>  accumulator + currentValue);

Reduce is an array iterator, which means that it goes over each element of the array and perform an action.
In this case:

**1st Iteration**:
accumulator - 0
currentValue - 1 (1st element of the array)
action: 0 + 1 ( so the accumulator will now be equal to 1)

**2nd Iteration**:
accumulator - 1
currentValue - 2 (2nd element of the array)
action: 1 + 2 ( so the accumulator will now be equal to 3)

**3rd Iteration**:
accumulator - 3
currentValue - 3 (2rd element of the array)
action: 3 + 3 ( so the accumulator will now be equal to 6)

Does it make sense?
We're summing them because we wrote as accumulator + currentValue, but we could be multiplying or subtracting, if that makes sense.